pkg/scale/cockroach_statefulset_test.go

```diff
@@ -0,0 +1,167 @@
+		tracker := cltSet.Tracker()
+		gvr := action.GetResource()
+		ns := action.GetNamespace()
+		verb := action.GetVerb()
+
+		switch verb {
+		case "get":
+			getAction := action.(clienttesting.GetAction)
+			obj, err := tracker.Get(gvr, ns, getAction.GetName())
+			require.NoError(t, err)
+
+			return true, obj, nil
+		case "list":
+			obj, err := tracker.List(gvr, schema.GroupVersionKind{
+				Group:   "",
+				Version: "v1",
+				Kind:    "Pod",
+			},
+				ns) // Tracker ignores labels passed in and only checks the namespace
+			require.NoError(t, err)
+
+			return true, obj, nil
+		default:
+			t.Logf("Unexpected action %v", action)
+			t.Fail()
+		}
+
+		return false, nil, nil
+	})
+
+	// No replicas running
+	require.Errorf(t, StatefulSetIsRunning(context.TODO(), cltSet, "crdb", "crdb-sts"), "statefulset replicas not yet reconciled. have %d expected %d",
+		sts.Status.Replicas,
+		sts.Spec.Replicas)
+
+	// Set Status Replicas Ready to be number of replicas defined in the spec
+	sts.Status.Replicas = stsReplicas
+	cltSet.Tracker().Update(sts.GroupVersionKind().GroupVersion().WithResource("statefulsets"), &sts, "crdb")
+
+	addPodsToStatefulSet(stsReplicas, sts, cltSet)
+
+	require.NoError(t, StatefulSetIsRunning(context.TODO(), cltSet, "crdb", "crdb-sts"))
+}
+
+func TestIsStatefulSetReadyToServe(t *testing.T) {
+	var stsReplicas int32
+	stsReplicas = 3
+
+	sts := statefulSet(stsReplicas)
+
+	cltSet := fakeclient.NewSimpleClientset()
+	require.NoError(t, cltSet.Tracker().Add(&sts))
+
+	cltSet.PrependReactor("*", "*", func(action clienttesting.Action) (handled bool, ret runtime.Object, err error) {
+		tracker := cltSet.Tracker()
+		gvr := action.GetResource()
+		ns := action.GetNamespace()
+		verb := action.GetVerb()
+
+		switch verb {
+		case "get":
+			getAction := action.(clienttesting.GetAction)
+			obj, err := tracker.Get(gvr, ns, getAction.GetName())
+			require.NoError(t, err)
+
+			return true, obj, nil
+		default:
+			t.Logf("Unexpected action %v", action)
+			t.Fail()
+		}
+
+		return false, nil, nil
+	})
+
+	// No replicas running
+	require.Error(t, IsStatefulSetReadyToServe(context.TODO(), cltSet, "crdb", "crdb-sts", stsReplicas))
+
+	// Set Status Replicas Ready to be number of replicas defined in the spec
+	sts.Status.Replicas = stsReplicas
+	cltSet.Tracker().Update(sts.GroupVersionKind().GroupVersion().WithResource("statefulsets"), &sts, "crdb")
+	require.NoError(t, IsStatefulSetReadyToServe(context.TODO(), cltSet, "crdb", "crdb-sts", stsReplicas))
+
+	// Err if there are more replicas running than in the spec
+	sts.Status.Replicas = stsReplicas * 2
+	cltSet.Tracker().Update(sts.GroupVersionKind().GroupVersion().WithResource("statefulsets"), &sts, "crdb")
+	require.Error(t, IsStatefulSetReadyToServe(context.TODO(), cltSet, "crdb", "crdb-sts", stsReplicas))
+
+}
+
+func statefulSet(stsReplicas int32) appsv1.StatefulSet {
```

Comment:
If we can reuse this code we need it in the test utils pkg

Reply:
I agree, I can refactor in a followup PR but don't want to keep this initial template to be out.

Reply:
Please file an issue

Reply:
https://github.com/cockroachdb/cockroach-operator/issues/562 done